arch/src/x86_64/mod.rs

```diff
@@ -672,6 +672,23 @@ pub fn generate_common_cpuid(
         });
     }
 
+    // Forward host L2 cache info.
+    // The Function 0x2 and 0x4 are related to cache. They are all the same value with KVM_GET_SUPPORTED_CPUID.
+    // Only function 0x8000_0006 is returned as 0.
+    for i in [0x8000_0006] {
+        cpuid.retain(|c| c.function != i);
+        if let Some(leaf) = get_host_cpu_cache_info(i) {
+            cpuid.push(CpuIdEntry {
+                function: i,
+                eax: leaf.eax,
+                ebx: leaf.ebx,
+                ecx: leaf.ecx,
+                edx: leaf.edx,
+                ..Default::default()
+            });
+        }
+    }
+
```

Comment:
```suggestion
    // Forward host L2 cache info.
    for i in [0x8000_0006] {
        // Check that this a valid CPUID leaf
        if unsafe { std::arch::x86_64::__cpuid(0x8000_0000)}.eax < i {
            break;
        }
        cpuid.retain(|c| c.function != i);
        let leaf = unsafe { std::arch::x86_64::__cpuid(i) };
        cpuid.push(CpuIdEntry {
              function: i,
              eax: leaf.eax,
              ebx: leaf.ebx,
              ecx: leaf.ecx,
              edx: leaf.edx,
              ..Default::default()
          });
        }
    }

```

Reply:
fixed

Reply:
Looks like you didn't pus the updated version?

Reply:
I fixed it like this:
before:
```rust
cpuid.retain(|c| c.function != I);
if let Some(leaf) = get_host_cpu_cache_info(i) {
    cpuid.push(CpuIdEntry {...});
}
```
after:
```rust
if let Some(leaf) = get_host_cpu_cache_info(i) {
    cpuid.retain(|c| c.function != I);
    cpuid.push(CpuIdEntry {...});
}
```